src/opencl/mod.rs

```diff
@@ -7,14 +7,23 @@ use std::fmt::Write;
 use std::hash::{Hash, Hasher};
 
 pub type BusId = u32;
+
+#[allow(non_camel_case_types)]
 pub type cl_device_id = ocl::ffi::cl_device_id;
 
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub enum Brand {
     Amd,
+    Apple,
     Nvidia,
 }
 
+impl Brand {
+    fn all() -> Vec<Brand> {
+        vec![Brand::Nvidia, Brand::Amd, Brand::Apple]
```

Comment:
could be an Array

Reply:
I considered. I'm not crazy about hard-coding the size into the signature. This expresses the intent more cleanly and doesn't need to be hyper-optimized.